File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/human-resources/department/department-lookup-model.ts
import { IEmployeeDepartmentHistoryLookupModel } from 'app/models/data/entities/human-resources/employee-department-history/employee-department-history-lookup-model';

export interface IDepartmentLookupModel {
  departmentID: number;
  name: string;
  groupName: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class DepartmentLookupModel implements IDepartmentLookupModel {
  departmentID: number;
  name: string;
  groupName: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<DepartmentLookupModel>) {
    if (!!init) {
      Object.assign<DepartmentLookupModel, Partial<DepartmentLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/HumanResources/Department/Commands/Queries/GetDepartments/DepartmentLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.HumanResources;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.HumanResources.Department.Queries.GetDepartments
{
    public partial class DepartmentLookupModel : IHaveCustomMapping
    {
        public short DepartmentID { get; set; }
        public string Name { get; set; }
        public string GroupName { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.HumanResources.Department, DepartmentLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<DepartmentLookupModel, BaseDepartment>().IgnoreMissingDestinationMembers();
        }
    }
}
